### PR TITLE
Delete separate class Controller

### PR DESCRIPTION
**Pull request by spring-zip**

### CODE REVIEW

File: src/com/myCompany/WA/repository/CitiesRepository.java [hunk 1]
```diff
@@ -33,13 +33,15 @@ public City getCityByIndex(int index) throws WeatherProgramException {
     }
 
     public City getCityByName(String cityName) throws WeatherProgramException {
-        if (!isCityInList(cityName)) {
+        if (isCityInList(cityName)) {
+            int index;
+            index = getIndexInLocalBase(cityName);
```
Comment (JamesVynn) [inline src/com/myCompany/WA/repository/CitiesRepository.java]: refactor and rename, getIndexInLocalBase  method must return city

File: src/com/myCompany/WA/program/WeatherProgram.java [hunk 2]
```diff
@@ -1,37 +1,48 @@
...
 
         while (true) {
             view.start();
 
-            if (controller.inputData().hasNextInt()) {
-                int historyNumber = controller.inputData().nextInt() - 1;
-
-                view.showCityWeather(controller.getCityByIndex(historyNumber));
+            if (input.hasNextInt()) {
+                int historyNumber = input.nextInt() - 1;
```
Comment (JamesVynn) [inline src/com/myCompany/WA/program/WeatherProgram.java]: why here is -1 ?

File: src/com/myCompany/WA/program/WeatherProgram.java [hunk 3]
```diff
@@ -1,37 +1,48 @@
...
 
+                try {
```
Comment (JamesVynn) [inline src/com/myCompany/WA/program/WeatherProgram.java]: move try catch to separate method

File: src/com/myCompany/WA/program/WeatherProgram.java [hunk 4]
```diff
@@ -1,37 +1,48 @@
...
                 if(inputString.equals(EXIT)){
                     break;
                 }
 
                 if (inputString.equals(HISTORY))  {
-                    view.showHistoryRequestCities(controller.getHistoryRequestCities());
+                    view.showHistoryRequestCities(repository.getListCities());
                 } else {
-                    view.showCityWeather(controller.getCityByName(inputString));
+                    try {
```
Comment (JamesVynn) [inline src/com/myCompany/WA/program/WeatherProgram.java]: move try catch to separate method

File: src/com/myCompany/WA/program/WeatherProgram.java [hunk 5]
```diff
@@ -1,37 +1,48 @@
...
 
     private static final String EXIT = "finish";
     private static final String HISTORY = "list";
 
     public void start() {
 
+        CitiesRepository repository = new CitiesRepository();
         WeatherView view = new WeatherView();
-        Controller controller = new Controller();
+        Scanner input = new Scanner(System.in);
```
Comment (JamesVynn) [inline src/com/myCompany/WA/program/WeatherProgram.java]: Scanner input should be in view

File: src/com/myCompany/WA/program/WeatherProgram.java [hunk 6]
```diff
@@ -1,37 +1,48 @@
...
 
         while (true) {
             view.start();
 
-            if (controller.inputData().hasNextInt()) {
-                int historyNumber = controller.inputData().nextInt() - 1;
-
-                view.showCityWeather(controller.getCityByIndex(historyNumber));
+            if (input.hasNextInt()) {
```
Comment (JamesVynn) [inline src/com/myCompany/WA/program/WeatherProgram.java]: analyse what types of commands do you have, create extra class to represent the command and then use it with switch

File: src/com/myCompany/WA/program/WeatherProgram.java (hunk 2 above)
Comment (spring-zip) [reply src/com/myCompany/WA/program/WeatherProgram.java]: For user history number begin at 1, in DB first index 0, from that made -1.

File: src/com/myCompany/WA/repository/CitiesRepository.java [hunk 7]
```diff
@@ -33,29 +33,29 @@ public City getCityByIndex(int index) throws WeatherProgramException {
...
     }
 
-    private int getIndexInLocalBase (String cityName) throws IndexOutOfBoundsException {
+    private City getCityInLocalBase(String cityName) {
         int index = -1;
+
         for (int i = 0; i < listCities.size(); i++) {
             City city = listCities.get(i);
             if (city.getCityName().equals(cityName)) {
                 index = i;
```
Comment (JamesVynn) [inline src/com/myCompany/WA/repository/CitiesRepository.java]: if you found city just return it, the rest of unneeded code to be removed

**Pull request merged**